src/System.Security.Principal.Windows/tests/WindowsIdentityTests.cs

```diff
@@ -57,6 +58,56 @@ public static void ConstructorsAndProperties()
     }
 
     [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public static void AuthenticationCtor(bool authentication)
+    {
+        SafeAccessTokenHandle token = WindowsIdentity.GetCurrent().AccessToken;
+        bool gotRef = false;
+        try
+        {
+            token.DangerousAddRef(ref gotRef);
+            IntPtr logonToken = token.DangerousGetHandle();
+
+            WindowsIdentity windowsIdentity = new WindowsIdentity(logonToken, authenticationType, WindowsAccountType.Normal, isAuthenticated: authentication);
```

Comment:
The code block between `IntPtr logonToken = token.DangerousGetHandle();` and `CheckDispose(windowsIdentity);` seems to be the only difference between the three tests here.

To reduce duplicate code, perhaps you could create a private method that does the common things between the three test methods. Then use and pass a lambda function to pass down the statements where you instantiate the ctor and verify values with assert calls.